chain/client/src/view_client.rs

```diff
@@ -446,14 +417,46 @@ impl Actor for ViewClientActor {
 }
 
 impl Handler<Query> for ViewClientActor {
-    type Result = Result<Option<QueryResponse>, String>;
+    type Result = Result<QueryResponse, QueryError>;
 
     #[perf]
     fn handle(&mut self, msg: Query, _: &mut Self::Context) -> Self::Result {
         self.handle_query(msg)
     }
 }
 
+#[derive(thiserror::Error, Debug)]
+#[error(transparent)]
+struct RuntimeQueryError(#[from] pub near_chain::near_chain_primitives::error::QueryError);
```

Comment:
0. Consider removing this wrapper (see my other comment in the `view_client.rs`)
1. If we keep it, let's document it
2. If we keep it, let's rename the struct to be more explicit

```suggestion
/// This is a helper struct which is used to implement `From<near_chain_primitives::error::QueryError>`
/// for `near_client_primitives::types::QueryError` (both are foreign types) to be able to use `?` operator.
#[derive(thiserror::Error, Debug)]
#[error(transparent)]
struct WrappedChainQueryError(#[from] pub near_chain::near_chain_primitives::error::QueryError);
```